src/data/list/basic.lean

```diff
@@ -1530,8 +1530,38 @@ theorem prod_hom [monoid β] (l : list α) (f : α → β) [is_monoid_hom f] :
 by { simp only [prod, foldl_map, (is_monoid_hom.map_one f).symm],
   exact l.foldl_hom _ _ _ 1 (is_monoid_hom.map_mul f) }
 
+-- `to_additive` chokes on this, so we do it by hand below
+@[simp]
+lemma prod_take_mul_prod_drop :
+  ∀ (L : list α) (i : ℕ), (L.take i).prod * (L.drop i).prod = L.prod
```

Comment:
If you generalise to arbitrary monoids, maybe to_additive becomes happy again?

Reply:
This already is an arbitrary `[monoid α]`. I wonder if the problem is that I match on `i = 0`.

Reply:
Aah, sorry. I was to fast. :man_facepalming: